include/openenclave/bits/evidence.h

```diff
@@ -25,6 +24,12 @@ OE_EXTERNC_BEGIN
     }
 #define X509_OID_FOR_OE_EVIDENCE_STRING "1.2.840.113556.10.1.2"
 
+#define X509_OID_FOR_NEW_OE_EVIDENCE_EXT                     \
+    {                                                        \
+        0x2B, 0x06, 0x01, 0x04, 0x01, 0x82, 0x37, 0x69, 0x02 \
+    }
+#define X509_OID_FOR_NEW_OE_EVIDENCE_STRING "1.3.6.1.4.1.311.105.2"
```

Comment:
Just a question, why do we need both format?

cc @yentsanglee

Reply:
The binary form to pass in to `mbedtls_x509write_crt_set_extension` to write to a certificate.
The string form to be compared with when parsing the certificate extension.

Reply:
Is there any need for supporting such conversion? I'm asking because I implemented a simple oid parsing function (hex to string) when porting OpenSSL-based OE crypto APIs #3395  (https://github.com/openenclave/openenclave/pull/3395/files#diff-b1a9d4b221a2f293e70eee3421a9aa6dR1087).
This is just for supporting OpenSSL version of `oe_gen_custom_x509_cert`. Can you take a look?

Reply:
@mingweishih if oe_gen_custom_x509_cert() takes oid string as input, we don't need to provide the binary forms. That will be great.

Reply:
Does that imply that the binary form is only needed by the mbedtls API (i.e., is the string format more common)? In that case, does it make sense to have `ext_oid` in the `oe_cert_config_t` to store the string format directly, and have the conversion function implemented in the mbedtls-based crypto layer instead of openssl-based one?

Reply:
The binary format is solely for the mbedtls API. The string format is more common and readable. However, the reason for having 2 forms is to eliminate the need for conversion. In your PR #3395, I see that you have to call _decode_oid_to_str() to convert to string. If that's the case for mbedtls too, then I'd suggest to have `ext_oid` to store the string form t avoid conversion.

Reply:
I think I'm referring to the consistency of our API regardless which crypto libraries is used. Do we have any API that takes oid as input? What I only know it's used internally in the `oe_generate_attestation_certificate` function, which I think it's fine to use hardcoded value there. I'm wondering if there is other case.

Reply:
There is no public APIs that take OID as input. All OID definitions are internal (either in `internal/evidence.h` or `internal/report.h`).

Reply:
Thanks for the clarification. Then, I think it makes sense to not doing the conversion by ourselves.